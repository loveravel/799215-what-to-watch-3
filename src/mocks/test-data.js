export const films = [
  {
    id: 1,
    name: `Movie 1`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: true
  },
  {
    id: 2,
    name: `Movie 2`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideoLink: ``,
    video: ``,
    isFavorite: false
  },
  {
    id: 3,
    name: `Movie 3`,
    genre: `Drama`,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: false
  },
];

export const movie = {
  id: 1,
  name: `Movie 1`,
  genre: ``,
  released: 2019,
  backgroundImage: ``,
  backgroundColor: ``,
  previewImage: ``,
  posterImage: ``,
  description: ``,
  rating: 4.5,
  scoresCount: 300,
  runTime: 3,
  director: ``,
  starring: [],
  previewVideo: ``,
  video: ``,
  isFavorite: false
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const userData = {
  id: 1,
  name: ``,
  email: ``,
  avatarUrl: ``
};
