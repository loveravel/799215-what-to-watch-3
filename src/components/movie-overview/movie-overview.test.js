import React from "react";
import renderer from "react-test-renderer";

import {movie} from "../../mocks/test-data.js";

import MovieOverview from "./movie-overview.jsx";

it(`<MovieOverview /> is rendered correctly`, () => {
  const props = {
    movie,
  };

  const tree = renderer
    .create(<MovieOverview {...props}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
