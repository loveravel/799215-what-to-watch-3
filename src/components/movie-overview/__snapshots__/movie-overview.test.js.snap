// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieOverview /> is rendered correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      9.5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very Good
      </span>
      <span
        className="movie-rating__count"
      >
        15
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    Having received a small factory from his father,
    Howard Hughes turned it into a gigantic, fantastically profitable enterprise.
    Having become the owner of a huge film company, he shot the most expensive
    film of his time and won the hearts of the most beautiful Hollywood actresses.
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Martin Scorsese
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Leonardo DiCaprio, Cate Blanchett, Kate Beckinsale, John C. Reilly, Alec Baldwin, Alan Alda, Ian Holm, Danny Huston, Gwen Stefani, Jude Law
      </strong>
    </p>
  </div>,
]
`;
